src/storage/src/storage_state.rs

```diff
@@ -1021,13 +1022,26 @@ impl<'w, A: Allocate> Worker<'w, A> {
                         }
                     }
                 }
-                StorageCommand::InitializationComplete
-                | StorageCommand::UpdateConfiguration(_)
-                | StorageCommand::CreateSinks(_) => (),
+                StorageCommand::CreateSinks(exports) => {
+                    // Ensure that exports are forward-rolling alter compatible.
+                    for export in exports {
+                        let prev = running_exports_descriptions
+                            .insert(export.id, export.description.clone());
+
+                        if let Some(prev_export) = prev {
+                            prev_export
+                                .alter_compatible(export.id, &export.description)
+                                .expect("only alter compatible ingestions permitted");
+                        }
```

Comment:
Untested

Reply:
This gets exercised during a full zippy run, but it would be nice if we had a better set of reconciliation tests.